include/gpio_wrapper.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "nrf_gpio.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+
+namespace edge::aidan {
+
+inline bool read_gpio(uint32_t pin_number)
+{
+    bool val = nrf_gpio_pin_read(pin_number);
+    return val;
+}
+
+inline void set_gpio(uint32_t pin_number)
+{
+    nrf_gpio_pin_set(pin_number);
+}
+
+inline void clear_gpio(uint32_t pin_number)
+{
+    nrf_gpio_pin_clear(pin_number);
+}
+
+inline void toggle_gpio(uint32_t pin_number)
+{
+    nrf_gpio_pin_toggle(pin_number);
+}
+
+inline void set_gpio_output(uint32_t pin_number)
+{
+    nrf_gpio_cfg_output(pin_number);
+}
+
+inline void set_gpio_input(uint32_t pin_number)
+{
+    nrf_gpio_cfg_input(pin_number, NRF_GPIO_PIN_NOPULL);
```

Comment:
wouldn't we want to expose the pull vs nopull somehow

Reply:
i think this is why its nrf_gpio_cfg_output/input instead of one function with direction passed

Reply:
yea mb forgot to add it

Reply:
done